relay-general/src/store/transactions/processor.rs

```diff
@@ -57,7 +57,13 @@ impl<'r> TransactionsProcessor<'r> {
 
                 if let Some((rule, result)) = result {
                     if *transaction != result {
-                        meta.set_original_value(Some(transaction.clone()));
+                        // If another rule was applied before, we don't want to
+                        // rename the transaction name to keep the original one.
+                        // We do want to continue adding remarks though, in
+                        // order to keep track of all rules applied.
+                        if meta.original_value().is_none() {
+                            meta.set_original_value(Some(transaction.clone()));
+                        }
```

Comment:
I tried to do this with `get_or_insert_with`, but I couldn't figure out how. Happy to change it.